doc/administrator/web/index.md

```diff
@@ -98,6 +98,21 @@ The main configuration parameters you will need to change are:
   * `ncbi.upload.namespace` - Prefix for file upload identifiers to NCBI. The namespace is used to guarantee upload IDs are unique.  This configuration option is used as a placeholder and may still be set by the user.
 5. **Security configuration**
  * `security.password.expiry` - The number of days a password is valid for in IRIDA.  After a password expires the user will be required to create a new one.  Passwords cannot be reused.
+6. **OAuth2 JWK security configuration** - IRIDA uses JWT (Json Web Tokens) for OAuth2 and as such requires a JWK (Json Web Key) in RSA format to encrypt and decrypt these tokens.
+ * `oauth2.jwk.rsakey.id=CHANGEME` - The id for the RSA private and public keys (used to select the correct keys for encryption and decryption)
+ * `oauth2.jwk.rsakey.private=/etc/irida/jwk-private.pem` - The RSA private key 
+ * `oauth2.jwk.rsakey.public=classpath:/jwk-public.pem` - The RSA public key
+
+### OAuth2 JWK RSA Key generation
+The RSA keys used to encrypt and decrypt the JWT's can be generated with the following commands:
```

Comment:
This is now required to install IRIDA, I'm not sure it makes sense to automate this. I had previously discussed with @Takadonet that it will need to be incorporated into `irida-puppet` in some manner.

These can be added to the `UPGRADING.md`, I was planning on adding it before our next release but if it is better to add it in this PR I could do that.

This is used to encrypt the access_token, since this PR changes to using JWT (Json Web Tokens). These tokens encode some user information, including what scopes have been authorized as part of this token. IRIDA uses this information to decode the User and the granted Authorities. Only IRIDA requires access to this Key, as the clients do not need to decode the information encoded in the access token.

Reply:
Fixed in [bc31bfd](https://github.com/phac-nml/irida/pull/1339/commits/bc31bfd2da4bd29e9dd7ff503b9921c866ff6381)

Reply:
Thanks for the explanation. Our discussions offline led us to the conclusion that automating the generation of keys as part of the IRIDA WAR wouldn't work in situations like multiple server deployments (at least not easily) so generating the keys separately makes sense.